### DISCUSSION TITLE
Skip installation of active slots but allow them in the bundle

### DISCUSSION THREAD

**Discussion by JSydll** [rauc/rauc#1005] [Q&A] [answered]
I'm currently facing a system with three different slot classes (two asymmetric recoveries - _don't ask why_ - and one prod). One of the recovery slots will be active/booted when the `rauc install` process is triggered. Now there is a requirement that the system can be updated in a transparent manner, meaning that the technician shall not need to know which recovery slot requires an update.

An approach to enable this could be to **allow a bundle to contain images for the active/booted slot(s) - and simply skip them without error during installation**. As the rest of the process remains unaltered, I should still be protected against hazardous artifacts in the bundle (because the process will fail if the image class is not known to the system).

This is a bit related to https://github.com/rauc/rauc/issues/745 but I would like to leverage the established installation process as much as possible and not work around it with hooks.

What's your opinion on this? Would a (configurable) feature like this be useful?

**Comment by ejoerns** [comment, marked as answer]
This actually sounds like a very special case, yes. :wink: 

Maybe a more generic variant that faces the same issue is if you have *A + recovery* and want provide a bundle that both supports updating the *recovery* from *A* and updating *A* from *recovery*.

Currently I do not have any better idea than providing an option as you proposed. If I got it correctly, you suggest something like:

``` ini
[image.rootfs]
image=rootfs.img
ignore-if-active=true

[image.recovery]
image=recovery.img
ignore-if-active=true
```

for

``` ini
[slot.rootfs.0]
device=/dev/sda

[slot.recovery.0]
device=/dev/sdb
```

Right?

**Comment by JSydll** [comment]
Yes, exactly. Your described use case makes even more sense than mine.

So I guess, I'll need to checkout your contribution guidelines and figure out how to implement and test this?
